test/library.cpp

```diff
@@ -680,17 +680,32 @@ TEST_F(LibraryTest, filterByPublisher)
 
 TEST_F(LibraryTest, filterByName)
 {
-  EXPECT_FILTER_RESULTS(kiwix::Filter().name("wikibooks_de"),
+  EXPECT_FILTER_RESULTS(kiwix::Filter().name("wikibooks.de"),
     "An example ZIM archive"
   );
 
-  EXPECT_FILTER_RESULTS(kiwix::Filter().query("name:wikibooks_de"),
-    "An example ZIM archive"
+  // Parsing the query with `name:` prefix splits the token on the dot, as if it was 2 sentences.
+  // It creates a query "XNwikibook@1 PHRASE 2 XNde@2".
+  // I haven't found the syntax to no split on dot
+  EXPECT_FILTER_RESULTS(kiwix::Filter().query("name:wikibooks.de"),
+    /* no results */
+  );
+
+  EXPECT_FILTER_RESULTS(kiwix::Filter().name("wikibooks"),
+    /* no results */
   );
 
   EXPECT_FILTER_RESULTS(kiwix::Filter().query("wikibooks_de"),
```

Comment:
Shouldn't this be changed to `wikibooks.de`?

Reply:
Indeed. I have add/keep tests with `wikibooks` and `wikibooks.de`.